src/subscriptiondata.h

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SUBSCRIPTIONDATA_H
+#define SUBSCRIPTIONDATA_H
+
+#include <QObject>
+
+class SubscriptionData final : public QObject {
+  Q_OBJECT
+  Q_DISABLE_COPY_MOVE(SubscriptionData)
+
+  Q_PROPERTY(bool initialized READ initialized NOTIFY changed)
+
+  // Subscription
+  Q_PROPERTY(int createdAt READ createdAt CONSTANT)
+  Q_PROPERTY(int expiresOn READ expiresOn CONSTANT)
+  Q_PROPERTY(bool isCancelled READ isCancelled CONSTANT)
+  Q_PROPERTY(QString status READ status CONSTANT)
+  Q_PROPERTY(QString type READ type CONSTANT)
+
+  // Plan
+  Q_PROPERTY(int planAmount READ planAmount CONSTANT)
+  Q_PROPERTY(QString planCurrency READ planCurrency CONSTANT)
+  Q_PROPERTY(int planIntervalCount READ planIntervalCount CONSTANT)
+
+  // Payment
+  Q_PROPERTY(QString paymentType READ paymentType CONSTANT)
+  Q_PROPERTY(QString creditCardBrand READ creditCardBrand CONSTANT)
+  Q_PROPERTY(QString creditCardLast4 READ creditCardLast4 CONSTANT)
+  Q_PROPERTY(int creditCardExpMonth READ creditCardExpMonth CONSTANT)
+  Q_PROPERTY(int creditCardExpYear READ creditCardExpYear CONSTANT)
+
+ public:
+  SubscriptionData();
+  ~SubscriptionData();
+
+  // [[nodiscard]] bool fromJson(const QByteArray& json);
+  Q_INVOKABLE bool fromJson(const QByteArray& json);
```

Comment:
Only made invokable by QML for testing purposes so we can use it [here](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/3606/files#diff-02cecf306084b766f5c5eb93e830ec20d443d0ad48be5f481eea2dd6285de48dR142).